pkg/apis/apps/v1alpha1/apimanagerrestore_types.go

```diff
@@ -0,0 +1,104 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// APIManagerRestoreSpec defines the desired state of APIManagerRestore
+// +k8s:openapi-gen=true
+type APIManagerRestoreSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+	RestoreSource APIManagerRestoreSource `json:"restoreSource"`
+}
+
+// APIManagerRestoreSource defines the backup data restore source
+// configurability. It is a union type. Only one of the fields can be
+// set
+type APIManagerRestoreSource struct {
+	// +optional
+	// Restore data soure configuration
+	PersistentVolumeClaim *PersistentVolumeClaimRestoreSource `json:"persistentVolumeClaim,omitempty"`
```

Comment:
why not actual `PersistentVolumeClaim`

Reply:
Only a subset of functionality is provided